File: src/main/java/railgunDownloaderV4/components/youtubeUI/ChoosePath.java
/*
 * RAILGUN DOWNLOADER VERSION 4.0.0
 * ChoosePath.java
 */
package railgunDownloaderV4.components.youtubeUI;
import javax.swing.*;
import java.awt.*;
import java.util.Objects;
import java.io.File;

public class ChoosePath {
    private final YoutubeUI YoutubeUIContext;

    public ChoosePath(YoutubeUI YoutubeUIContext) {
        this.YoutubeUIContext = YoutubeUIContext;
    }
    public void SetChoosePath(JFrame App, JButton ChoosePathButton) {
        JFileChooser BrowserFolder = new JFileChooser();

        Image ButtonIcon = Toolkit.getDefaultToolkit().getImage(
                Objects.requireNonNull(getClass().getResource("/FolderPath.png"))
        );
        ChoosePathButton.setBorder(null);
        ChoosePathButton.setLocation(new Point(100, 420));
        ChoosePathButton.setSize(new Dimension(50, 50));
        ChoosePathButton.setContentAreaFilled(false);
        ChoosePathButton.setIcon(new ImageIcon(ButtonIcon.getScaledInstance(50, 50, Image.SCALE_FAST)));
        ChoosePathButton.setBorderPainted(false);
        ChoosePathButton.setFocusPainted(false);

        ChoosePathButton.setToolTipText("Choose save path");

        OpenSaveFolderPath(App, BrowserFolder, ChoosePathButton);
        App.add(ChoosePathButton);
    }

    private void OpenSaveFolderPath(JFrame App, JFileChooser FolderBrowser, JButton ChoosePathButton) {
        ChoosePathButton.addActionListener(e -> {
            FolderBrowser.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);

            int OpenResult = FolderBrowser.showDialog(App, "Choose save path");

            if(OpenResult == JFileChooser.APPROVE_OPTION) {
                File ChoosePath = FolderBrowser.getSelectedFile();
                YoutubeUIContext.ShowPathField.setText(ChoosePath.toString());
            }
        });
    }
}

File: src/main/kotlin/railgunDownloaderV4/components/youtubeUI/ChoosePath.kt
/*
 * RAILGUN DOWNLOADER - VERSION 4.0.0
 * File: ChoosePath.kt
 * This project is licensed GPL-3.0 License.
 * LICENSE: https://www.gnu.org/licenses/gpl-3.0.html
 * Contribution: Reim
 */
package railgunDownloaderV4.components.youtubeUI

import railgunDownloaderV4.components.ulti.SetIconButton
import java.awt.Dimension
import java.awt.Point
import javax.swing.JButton
import javax.swing.JFileChooser
import javax.swing.JFrame

class ChoosePath(private val youtubeUI: YoutubeUI) {

    private val setIconButton: SetIconButton by lazy { SetIconButton() }

    fun setChoosePath(app: JFrame, choosePathButton: JButton) {
        val browserFolder = JFileChooser()
        choosePathButton.border = null
        choosePathButton.location = Point(100, 420)
        choosePathButton.size = Dimension(50, 50)
        choosePathButton.isContentAreaFilled = false
        choosePathButton.isBorderPainted = false
        choosePathButton.isFocusPainted = false
        setIconButton.setIcon(
            choosePathButton, "/FolderPath.png",
            50, 50
        )
        choosePathButton.toolTipText = "Choose save directory path"

        openSaveDir(browserFolder, choosePathButton)
        app.add(choosePathButton)
    }

    private fun openSaveDir(folderBrowser: JFileChooser, choosePathButton: JButton) {
        choosePathButton.addActionListener {
            folderBrowser.fileSelectionMode = JFileChooser.DIRECTORIES_ONLY
            folderBrowser.dialogTitle = "Choose save directory path"

            folderBrowser.takeIf { it.showOpenDialog(null) == JFileChooser.APPROVE_OPTION }?.selectedFile?.let {
                youtubeUI.pathField.text = it.absolutePath
            }
        }
    }
}
